### PR TITLE
[ci][byod/2] move 9 more tests to byod

### PR DESCRIPTION
**Pull request by can-anyscale**
## Why are these changes needed?
Move 9 more release tests to byod

For context on why byod is preferrable: https://docs.google.com/document/d/12WU4vpR0jHgOwceoSgPtE4B-B-6mBonnHHiCSMmMNoY/edit

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- Testing Strategy
   - [X] Unit tests
   - [X] Release tests - https://buildkite.com/ray-project/release-tests-pr/builds/40886

### CODE REVIEW

File: release/release_tests.yaml [hunk 1]
```diff
@@ -4524,8 +4549,13 @@
   working_dir: nightly_tests
 
   frequency: nightly
+  python: "3.8"
   team: core
   cluster:
+    byod:
+      type: cpu
```
Comment (aslonnie) [inline release/release_tests.yaml:4556]: should we just make `type: cpu` the default type?

Comment (aslonnie) [reply release/release_tests.yaml:4556]: and support `byod: ~`

Comment (can-anyscale) [reply release/release_tests.yaml:4556]: I like that!

**Comment by ollie-iterators** [comment]
Speaking of the release tests, I've always been wondering, why do the release tests not run as much as the other tests? Does that mean that they never get removed from flakey-tests.ray.io?

**Comment by aslonnie** [comment]
> why do the release tests not run as much as the other tests?

because they are relatively expensive to run.

> Does that mean that they never get removed from flakey-tests.ray.io?

not sure what do you mean..

**Comment by ollie-iterators** [comment]
> > why do the release tests not run as much as the other tests?
> 
> because they are relatively expensive to run.
> 
> > Does that mean that they never get removed from flakey-tests.ray.io?
> 
> not sure what do you mean..

The tests shown in the image below on https://flakey-tests.ray.io/ with "Show Release Tests" set to Only and "Show All" set to true  have no failures and are still shown as flakey tests.

File: release/ray_release/byod/byod.Dockerfile [hunk 2]
```diff
@@ -18,4 +18,5 @@ sudo apt-get update -y \
 EOF
 
 COPY "$PIP_REQUIREMENTS" .
+RUN "$HOME"/anaconda3/bin/pip install --no-cache-dir https://ray-ci-deps-wheels.s3.us-west-2.amazonaws.com/AutoROM.accept_rom_license-0.5.4-py3-none-any.whl
```
Comment (aslonnie) [inline release/ray_release/byod/byod.Dockerfile:21]: how big is this thing? do we need this for most of the tests?

Comment (can-anyscale) [reply release/ray_release/byod/byod.Dockerfile:21]: It's 500kb, only a few test needs this. This should have gone into requirements_byod but it doesn't not have a pip version

**Review by aslonnie — APPROVED** [review]

**Pull request merged**